Source/engine/render/text_render.cpp

```diff
@@ -411,15 +418,21 @@ uint32_t DoDrawString(const Surface &out, string_view text, Rectangle rect, Poin
 		if (next == ZWSP)
 			continue;
 
-		const uint32_t unicodeRow = GetUnicodeRow(next);
+		uint32_t unicodeRow = GetUnicodeRow(next);
 		if (unicodeRow != currentUnicodeRow || font == nullptr) {
 			kerning = LoadFontKerning(size, unicodeRow);
-			font = LoadFont(size, color, unicodeRow);
+			font = &LoadFont(size, color, unicodeRow);
+			if (!*font) {

```

Comment:
This doesn't quite make sense. `*font` is a `const OwnedClxSpriteList&`. Applying an implicit boolean conversion to it is not meaningful.

Reply:
Oh nvm `font` is a pointer here.

Reply:
It's been changed to `OptionalOwnedClxSpriteList` since the fail case was returning a `OwnedClxSpriteList` with null data and not a `nullptr`.

Reply:
nit: Let's use `OptionalClxSpriteList` instead of a pointer?

Reply:
Wasn't sure how to do so with out causing a copy. Or do you think doing so would be ok?

Reply:
`OptionalClxSpriteList` is free to copy, it's represented as a single pointer.
`OptionalClxSpriteList` ~ `OptionalOwnedClxSpriteList` relationship is similar to `std::optional<std::string_view>` ~ `std::optional<std::string>`.